src/elements/caprover/Caprover.wc.svelte

```diff
@@ -78,21 +92,89 @@
     message = undefined;
 
     deployCaprover(data, profile)
-      .then((data) => {
-        modalData = data;
-        deploymentStore.set(0);
-        success = true;
+      .then(async (vm) => {
+        if (data.workers.length == 0) {
+          success = true;
+          modalData = vm;
+          deploymentStore.set(0);
+        }
       })
       .catch((err: string) => {
         failed = true;
         message = err;
       })
-      .finally(() => {
-        loading = false;
+      .finally(async () => {
+        if (data.workers.length > 0) {  
+          for (const worker of data.workers) {
+            worker.publicKey = data.publicKey;
+
+            /* Docker disk */
+            const disk = new DiskModel();
+            disk.name = "data0";
+            disk.size = worker.diskSize;
+            disk.mountpoint = "/var/lib/docker";
+
+            const workerModel = new AddMachineModel();
+
+            workerModel.deployment_name = data.name;
+            workerModel.cpu = worker.cpu;
+            workerModel.memory = worker.memory;
+            workerModel.disks = [disk];
+            workerModel.node_id = worker.nodeId;
+            workerModel.public_ip = true;
+            workerModel.name = `CRW${worker.name}`;
+            workerModel.planetary = false;
+            workerModel.flist = CAPROVER_FLIST;
+            workerModel.qsfs_disks = [];
+            workerModel.rootfs_size = rootFs(worker.cpu, worker.memory);
+            workerModel.entrypoint = "/sbin/zinit init";
+            workerModel.env = {
+              SWM_NODE_MODE: "worker",
+              LEADER_PUBLIC_IP: data.publicIP,
+              CAPTAIN_IMAGE_VERSION: "latest",
+              PUBLIC_KEY: `${worker.publicKey}`,
+              CAPTAIN_IS_DEBUG: "true",
+            };
+            grid.machines
+            .add_machine(workerModel)
```

Comment:
when deploying the caprover leader and workers at the same time, we can do that in one step as they don't depend on each other now